txpipe/theory.py

```diff
@@ -103,81 +110,85 @@ def replace_with_theory_real(self, s, cosmo):
         ell = np.unique(np.logspace(np.log10(2), 5, 400).astype(int))
         tracers = self.get_ccl_tracers(s, cosmo)
 
-        for i in range(nbin_source):
-            for j in range(i + 1):
-                print(f"Computing theory lensing-lensing ({i},{j})")
+        if "galaxy_shear_xi_plus" and "galaxy_shear_xi_minus" in s.get_data_types():
+            for i in range(nbin_source):
+                for j in range(i + 1):
+                    print(f"Computing theory lensing-lensing ({i},{j})")
+
+                    # compute theory
+                    print(tracers[f"source_{i}"], tracers[f"source_{j}"])
+                    cl = pyccl.angular_cl(
+                        cosmo, tracers[f"source_{i}"], tracers[f"source_{j}"], ell
+                    )
+                    theta, *_ = s.get_theta_xi(
+                        "galaxy_shear_xi_plus", f"source_{i}", f"source_{j}"
+                    )
+                    xip = pyccl.correlation(cosmo, ell, cl, theta / 60, corr_type="L+")
```

Comment:
are the units right here? I have no idea what pyccl wants or what we're saving things in!  Worth a comment to clarify.

Reply:
CCL gets degrees in that function, so it should be correct. Added a comment.